sh.c

```diff
@@ -219,72 +226,84 @@ char char_bk;
 void print_escaped_text(text t, bool for_printf) {
   int i;
 
-  if (t == 0) return;
+  if (t != 0) {
```

Comment:
Took the opportunity to rewrite the `print_text` functions using a switch, which should be faster since it does less variables lookup in shell.